mlflow/langchain/api_request_parallel_processor.py

```diff
@@ -336,6 +365,58 @@ def _try_transform_response_to_chat_format(response):
         else:
             return transformed_response.model_dump(mode="json")
 
+    @staticmethod
+    def _try_transform_response_iter_to_chat_format(chunk_iter):
+        from langchain_core.messages.ai import AIMessageChunk
+
+        def _gen_converted_chunk(message_content, finish_reason):
+            transformed_response = _ChatChunkResponse(
+                id=None,
+                created=int(time.time()),
+                model=None,
+                choices=[
+                    _ChatChoiceDelta(
+                        index=0,
+                        delta=_ChatDeltaMessage(
+                            role="assistant",
+                            content=message_content,
+                        ),
+                        finish_reason=finish_reason,
+                    )
+                ],
+            )
+
+            if Version(pydantic.__version__) < Version("2.0"):
+                return json.loads(transformed_response.json())
+            else:
+                return transformed_response.model_dump(mode="json")
+
+        def _convert(chunk):
+            if isinstance(chunk, str):
+                message_content = chunk
+                finish_reason = None
+            elif isinstance(chunk, AIMessageChunk):
+                message_content = chunk.content
+                if hasattr(chunk, "response_metadata"):
+                    finish_reason = chunk.response_metadata.get("finish_reason")
+                else:
+                    finish_reason = None
+            elif isinstance(chunk, AIMessage):
+                # The langchain chat model does not support stream
+                # so `model.stream` returns the whole result.
+                message_content = chunk.content
+                finish_reason = "stop"
+            else:
+                return chunk
+            return _gen_converted_chunk(message_content, finish_reason=finish_reason)
+
+        def _result_gen_fn():
+            for chunk in chunk_iter:
+                converted_chunk = _convert(chunk)
+                yield converted_chunk
+
+        return _result_gen_fn()
```

Comment:
```suggestion
        for chunk in chunk_iter:
            converted_chunk = _convert(chunk)
            yield converted_chunk
```

we can remove this function.

Reply:
https://github.com/mlflow/mlflow/pull/11490/files#r1549329823